app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -102,258 +101,296 @@ public DataPullTask(String username, String password,
+                return;
             }
+            String sandboxId = PropertyUtils.genUUID().replace("-", "");
+            ukrForLogin = new UserKeyRecord(username, UserKeyRecord.generatePwdHash(password),
+                    newKey.getEncoded(), new Date(), new Date(Long.MAX_VALUE), sandboxId);
+        } else {
+            ukrForLogin = UserKeyRecord.getCurrentValidRecordByPassword(
+                    CommCareApplication._().getCurrentApp(), username, password, true);
+            if (ukrForLogin == null) {
+                Logger.log(AndroidLogger.TYPE_ERROR_ASSERTION,
+                        "Shouldn't be able to not have a valid key record when OTA restoring with a key server");
+            }
+        }
+    }
+
+    private static boolean shouldGenerateFirstKey() {
+        String keyServer = CommCarePreferences.getKeyServer();
+        return keyServer == null || keyServer.equals("");
+    }
+
+    /*
+     * @return the proper result, or null if we have not yet been able to determine the result to
+     * return
+     * @throws IOException
+     */
+    private ResultAndError<PullTaskResult> makeRequestAndHandleResponse(AndroidTransactionParserFactory factory)
+            throws IOException {
+
+        RemoteDataPullResponse pullResponse =
+                dataPullRequester.makeDataPullRequest(this, requestor, server, !loginNeeded);
+        Logger.log(AndroidLogger.TYPE_USER,
+                "Request opened. Response code: " + pullResponse.responseCode);
+
+        if (pullResponse.responseCode == 401) {
+            return handleAuthFailed();
+        } else if (pullResponse.responseCode >= 200 && pullResponse.responseCode < 300) {
+            return handleSuccessResponseCode(pullResponse, factory);
+        } else if (pullResponse.responseCode == 412) {
+            return handleBadLocalState(factory);
+        } else if (pullResponse.responseCode == 500) {
+            return handleServerError();
+        }
+        return null;
+    }
+
+    private ResultAndError<PullTaskResult> handleAuthFailed() {
+        wipeLoginIfItOccurred();
+        Logger.log(AndroidLogger.TYPE_USER, "Bad Auth Request for user!|" + username);
+        return new ResultAndError<>(PullTaskResult.AUTH_FAILED);
+    }
+
+    /**
+     * @return the proper result, or null if we have not yet been able to determine the result to
+     * return
+     * @throws IOException
+     */
+    private ResultAndError<PullTaskResult> handleSuccessResponseCode(
+            RemoteDataPullResponse pullResponse, AndroidTransactionParserFactory factory)
+            throws IOException {
+
+        handleLoginNeededOnSuccess();
+        this.publishProgress(PROGRESS_AUTHED, 0);
+
+        if (isCancelled()) {
+            // About to enter data commit phase; last chance to finish early if cancelled.
+            return new ResultAndError<>(PullTaskResult.UNKNOWN_FAILURE, "");
+        }
+
+        this.publishProgress(PROGRESS_DOWNLOADING_COMPLETE, 0);

```

Comment:
Hm, I just copied this over exactly from the existing code. You're saying you think it should be 100 instead of 0 anyway?